src/mon/OSDMonitor.cc

```diff
@@ -2135,6 +2135,9 @@ bool OSDMonitor::prepare_failure(MonOpRequestRef op)
 	dout(10) << " failure_info for osd." << target_osd << " now "
 		 << fi.reporters.size() << " reporters" << dendl;
       }
+      if (pending_inc.pending_osd_state_clear(target_osd, CEPH_OSD_UP)) {
```

Comment:
When looking more carefully at this situation, perhaps we should move this above into the  `if (fi.reporters.empty()) {` branch.

What do you think?